pkg/apis/pipeline/v1beta1/provenance.go

```diff
@@ -13,22 +13,20 @@ limitations under the License.
 
 package v1beta1
 
-// Provenance contains some key authenticated metadata about how a software artifact was
-// built (what sources, what inputs/outputs, etc.). For now, it only contains the subfield
-// `ConfigSource` that identifies the source where a build config file came from.
-// In future, it can be expanded as needed to include more metadata about the build.
-// This field aims to be used to carry minimum amount of the authenticated metadata in *Run status
-// so that Tekton Chains can pick it up and record in the provenance it generates.
+// Provenance contains metadata about resources used in the TaskRun/PipelineRun
+// such as the source from where a remote build definition was fetched.
+// This field aims to carry minimum amoumt of metadata in *Run status so that
+// Tekton Chains can capture them in the provenance.
 type Provenance struct {
-	// ConfigSource identifies the source where a resource came from.
-	ConfigSource *ConfigSource `json:"configSource,omitempty"`
+	// Source field identifies the source where a remote built script came from.
+	Source *Source `json:"source,omitempty"`
 }
 
-// ConfigSource identifies the source where a resource came from.
-// This can include Git repositories, Task Bundles, file checksums, or other information
-// that allows users to identify where the resource came from and what version was used.
-type ConfigSource struct {
-	// URI indicates the identity of the source of the config.
+// Source contains the information that can uniquely identify where a remote
+// built definition came from i.e. Git repositories, Tekton Bundles in OCI registry
+// and hub.
+type Source struct {
```

Comment:
I'm worried this is going to be confused as an input source type - This probably need to be clearer that it's for output. If we're not okay with the Config naming, perhaps we should renaming this to something like `RefProvenance`, or consider moving the fields in ConfigSource directly into provenance?

What were other fields we were considering adding to provenance?

Reply:
> I'm worried this is going to be confused as an input source type - This probably need to be clearer that it's for output.

that's a valid point. What about renaming the struct to be `RemoteSource`?  Similarly the field in provenance struct is also called `remoteSource`. 

> What were other fields we were considering adding to provenance?

Based on the discussion in Chains WG - Sep 29, 2022 ([meeting note](https://docs.google.com/document/d/1UVPSCDyNO-TzEFSv8jrqrEOF_FmV8NFuXncFm1gwmeY/edit#bookmark=id.rjj1d2yrtqfn)), we may extend this provenance field by adding fields like `input` and `output`, originally proposed in [TEP-109](https://github.com/tektoncd/community/blob/main/teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md#using-run-status-to-generate-provenance-metadata).

Reply:
+1 to not naming it as source.
RemoteSource has the same problem as source, how about RemoteResource?

Reply:
I feel like RemoteResource sounds more like the resource itself instead of the source information. What about `ResolvedSource`?

Reply:
Agree that RemoteResouce does not seem to suggest that it is source information.
ResolvedSource still feels like source code but resolved. 
Maybe ResourceSource conveys the information that this is the source of the resource but misses out on the fact that this source is remote.

Reply:
ResourceSource conveys the meaning but also a bit of alliteration :D 
In terms of provenance, one could argue that any resources that is not embedded in the `*Run` object is remote, including those stored on the cluster.

Reply:
From this bunch I like ResolvedSource or ResolverSource

Reply:
cc @bobcatfish for thoughts. Thanks!

Reply:
It's true that it's the same name as SLSA provenance, but it's only a name in SLSA v0.2 specification, and SLSA v1 will be very likely to not have a field named 'configSource'. 

That said, I feel like it's worth making this field more generic instead of attaching it to a specific SLSA version.

Reply:
Good to know! Then what is the name in V1?

Reply:
Sorry for the delay @Yongxuanzhang 

> Good to know! Then what is the name in V1?
It's likely to be part of [external parameters](https://github.com/MarkLodato/slsa/blob/03a066000033fad54af5ce5d88f8aa672b130237/docs/provenance/v1.0.proto#L56-L104).

> what we want to do in this PR is to seek a more generic name which works no matter how SLSA changes its' naming?
yes. I think that's the intention here.

Reply:
hey all! dropping in some ideas from a convo that @chuangw6 and I had about this today, pros and cons of some options for the name:

- If naming the one field isn't clear, maybe we could make a more complex object to make things clearer? isntead of `status.provenance.configsource` something like:
  - `status.provenance.task/pipeline.source` <-- not great to have a different field name depending on whether the run is a pipelinerun or taskrun 🙃 
  - status.provenance.tekton.source
- Config/ConfigSource
  - Con: not clear what "config" means in this context
  - Con: SLSA provenance format leaking into Tekton Pipelines API (and that format itself is shifting, e.g. the changes @chuangw6 has described around this field) / coupling Tekton Pipelines API to to SLSA provenance format
- Source
  - Con: Potential confusion for source used to build artifact
- RemoteSource & ResolvedSource & ResolverSource
  - Con: similar to Source (could be artifact source)
  - Pro: linked to remote resolution feature
- Build definition / build script
  - Pro: fits the way folks talk about 'builds'
  - Con: Pipeline/Task might not be building at all, could be just tests and other automation (and b/c of this 'build' is a bit of an anachronism)
- SpecSource / ResourceSource / DefinitionSource
- TektonSource

Maybe something like `TektonSource` could be a nice balance of being specific and not being confused with the source for the artifact?

Reply:
`TektonSource`

Reply:
`Source`

Reply:
`SpecSource`

Reply:
`TektonRemoteSource`

(to reflect that's from remote place.)

Reply:
I've changed the name to `RefSource` based on the discussion in https://github.com/tektoncd/pipeline/issues/6350. PTAL.